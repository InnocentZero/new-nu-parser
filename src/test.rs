use crate::resolver::Resolver;
use crate::{compiler::Compiler, parser::Parser};
use std::path::Path;
use std::process::exit;

fn evaluate_example(fname: &Path) -> String {
    let mut compiler = Compiler::new();
    let contents = std::fs::read(fname).expect("We only run tests found by glob");

    let span_offset = compiler.span_offset();
    compiler.add_file(&fname.to_string_lossy(), &contents);

    let parser = Parser::new(compiler, span_offset);
    compiler = parser.parse();

    let mut result = compiler.display_state();

    if !compiler.errors.is_empty() {
        return result;
    }

    let mut resolver = Resolver::new(&compiler);
    resolver.resolve();
    result.push_str(&resolver.display_state());

    result
}

#[test]
fn test_node_output() {
    insta::glob!("../tests", "*.nu", |path| {
        insta::assert_snapshot!(evaluate_example(path));
    });
}
