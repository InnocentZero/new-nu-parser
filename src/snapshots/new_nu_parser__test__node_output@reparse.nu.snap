---
source: src/test.rs
expression: "evaluate_example(path, &contents)"
input_file: tests/reparse.nu
---
0: Name (0 to 3)
1: Name (4 to 7)
2: Name (9 to 10)
3: Garbage (10 to 11)
4: Name (12 to 18)
5: List([NodeId(2), NodeId(3), NodeId(4)]) (8 to 18)
6: Name (22 to 26)
7: Variable (27 to 29)
8: Block(BlockId(0)) (20 to 31)
9: Name (33 to 36)
10: Int (37 to 38)
11: Block(BlockId(1)) (0 to 0)

